docs/current_docs/cookbook/snippets/custom-spans/go/main.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type MyModule struct{}
+
+func (m *MyModule) Foo(ctx context.Context) error {
+	// clone the source code repository
+	source := dag.
+		Git("https://github.com/dagger/hello-dagger").
+		Branch("main").Tree()
+
+	// list versions to test against
+	versions := []string{"20", "22", "23"}
+
+	// define errorgroup
+	eg, ctx := errgroup.WithContext(ctx)
```

Comment:
This doesn't work correctly atm, the spans and test results appear separately. See https://v3.dagger.cloud/dagger/traces/5a1baad5c3fd503a8901c9339a2ef911

Reply:
I pushed a fix.

Reply:
In this case I don't think we want `errgroup.WithContext` since that will result in interrupting the others when one fails.

```suggestion
	eg := new(errgroup.Group)
```

Reply:
Good point \cc @vikram-dagger